src/utils/selectors/index.ts

```diff
@@ -18,9 +18,29 @@ export const typedStore = `MethodDefinition[kind='constructor'] Identifier>TSTyp
 
 export const ngModuleDecorator = `ClassDeclaration > Decorator > CallExpression[callee.name='NgModule']`
 
-export const ngModuleProviders = `${ngModuleDecorator} ObjectExpression Property[key.name='providers'] > ArrayExpression Identifier`
-
-export const ngModuleImports = `${ngModuleDecorator} ObjectExpression Property[key.name='imports'] > ArrayExpression CallExpression[callee.object.name='EffectsModule'][callee.property.name=/forRoot|forFeature/] ArrayExpression > Identifier`
+export const ngModuleImports =
+  `${ngModuleDecorator} ObjectExpression ${metadataProperty(
+    'imports',
+  )} > ArrayExpression` as const
+
+export const ngModuleProviders =
+  `${ngModuleDecorator} ObjectExpression ${metadataProperty(
+    'providers',
+  )} > ArrayExpression` as const
+
+export const effectsInNgModuleImports =
+  `${ngModuleImports} CallExpression[callee.object.name='EffectsModule'][callee.property.name=/^for(Root|Feature)$/] ArrayExpression > Identifier` as const
+
+export const effectsInNgModuleProviders =
+  `${ngModuleProviders} Identifier` as const
+
+export function metadataProperty(key: RegExp): string
+export function metadataProperty<TKey extends string>(
+  key: TKey,
+): `Property:matches([key.name=${TKey}][computed=false], [key.value=${TKey}], [key.quasis.0.value.raw=${TKey}])`
+export function metadataProperty(key: RegExp | string): string {
```

Comment:
I noticed that you seem to prefer `arrow function` to `function`, however as it needs to be overloaded, I used `function` here.